test/osprovider.go

```diff
@@ -42,6 +43,59 @@ func downloadPuipui(destDir string) ([]string, error) {
 	return extract.Uncompress(resp.Filename, destDir)
 }
 
+func downloadFedora(destDir string) (string, error) {
+	const fedoraVersion = "40"
+	arch := kernelArch()
+	release := "1.14"
+	buildString := fmt.Sprintf("%s-%s-%s", arch, fedoraVersion, release)
+
+	var fedoraURL = fmt.Sprintf("https://download.fedoraproject.org/pub/fedora/linux/releases/%s/Cloud/%s/images/Fedora-Cloud-Base-AmazonEC2.%s.raw.xz", fedoraVersion, arch, buildString)
+
+	// https://github.com/cavaliergopher/grab/issues/104
+	grab.DefaultClient.UserAgent = "vfkit"
+	resp, err := grab.Get(destDir, fedoraURL)
+	if err != nil {
+		return "", err
+	}
+	return uncompressFedora(resp.Filename, destDir)
+}
+
+func uncompressFedora(fileName string, targetDir string) (string, error) {
+	file, err := os.Open(filepath.Clean(fileName))
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	stat, err := file.Stat()
+	if err != nil {
+		return "", err
+	}
+	header := make([]byte, min(262, stat.Size()))
+	if _, err := io.ReadFull(file, header); err != nil {
+		return "", err
+	}
```

Comment:
The header reading is still there? I've removed locally and the tests still build/pass
```
-       stat, err := file.Stat()
-       if err != nil {
-               return "", err
-       }
-       header := make([]byte, min(262, stat.Size()))
-       if _, err := io.ReadFull(file, header); err != nil {
-               return "", err
-       }
-       if _, err := file.Seek(0, 0); err != nil {
-               return "", err
-       }
-
```

Reply:
sorry my bad. I had a fight with vscode during stashing/rebasing. Yeah it's not necessary, I guess my first idea was to verify the file was actually the one I was expecting by checking the format and I saw a kind of example with this thing. Removed

Reply:
We use https://github.com/h2non/filetype in crc to do this kind of type checks. Here I'd assume the xz uncompressor will tell us if the file is not in the right format.